File: app/src/main/java/ru/blackbull/eatogether/ProfileActivity.java
package ru.blackbull.eatogether;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.AdapterView;
import android.widget.ArrayAdapter;
import android.widget.ListView;

import androidx.appcompat.app.AppCompatActivity;
import androidx.core.view.GravityCompat;
import androidx.drawerlayout.widget.DrawerLayout;
import androidx.fragment.app.FragmentTransaction;

import com.google.firebase.auth.FirebaseAuth;
import com.google.firebase.auth.FirebaseUser;

public class ProfileActivity extends AppCompatActivity implements View.OnClickListener, AdapterView.OnItemClickListener {
    private String[] titles;
    private DrawerLayout drawerLayout;
    private ListView drawerList;
    private FirebaseAuth mAuth;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_profile);
        mAuth = FirebaseAuth.getInstance();
        titles = getResources().getStringArray(R.array.titles);

        drawerLayout = findViewById(R.id.profile_drawer_layout);
        drawerList = findViewById(R.id.profile_drawer);
        drawerList.setAdapter(new ArrayAdapter<>(this, android.R.layout.simple_list_item_1, titles));
        drawerList.setOnItemClickListener(this);

        FragmentTransaction ft = getSupportFragmentManager().beginTransaction();
        ft.replace(R.id.profile_fragment_container, new ProfileFragment());
        ft.setTransition(FragmentTransaction.TRANSIT_FRAGMENT_FADE);
        ft.commit();
    }

    @Override
    protected void onStart() {
        super.onStart();
        FirebaseUser user = mAuth.getCurrentUser();
        if (user == null) {
            Intent intent = new Intent(getApplication(), WelcomeActivity.class);
            startActivity(intent);
        }
    }

    @Override
    public void onClick(View view) {
        switch (view.getId()) {
            case R.id.profile_settings:
                Intent intent = new Intent(getApplication(), EditProfileActivity.class);
                startActivity(intent);
                break;
            case R.id.profile_menu_btn:
                drawerLayout.openDrawer(GravityCompat.START);
                break;
            default:
                break;
        }
    }

    @Override
    public void onBackPressed() {
        if (drawerLayout.isDrawerOpen(drawerList)) {
            drawerLayout.closeDrawer(drawerList);
        } else {
            super.onBackPressed();
        }
    }

    @Override
    public void onItemClick(AdapterView<?> adapterView, View view, int i, long l) {
//        Fragment fragment = new ProfileFragment();
//        FragmentTransaction
        drawerLayout.closeDrawer(drawerList);
    }
}

File: app/src/main/java/ru/blackbull/eatogether/ui/ProfileActivity.kt
package ru.blackbull.eatogether.ui

import android.content.Intent
import android.os.Bundle
import android.util.Log
import android.view.View
import android.widget.AdapterView
import android.widget.AdapterView.OnItemClickListener
import android.widget.ArrayAdapter
import android.widget.ListView
import android.widget.Toast
import androidx.activity.viewModels
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.GravityCompat
import androidx.drawerlayout.widget.DrawerLayout
import androidx.fragment.app.FragmentTransaction
import com.google.firebase.auth.FirebaseAuth
import ru.blackbull.eatogether.R
import ru.blackbull.eatogether.ui.fragments.EditProfileFragment
import ru.blackbull.eatogether.ui.fragments.ProfileFragment
import ru.blackbull.eatogether.ui.viewmodels.FirebaseViewModel

class ProfileActivity : AppCompatActivity() , View.OnClickListener ,
    OnItemClickListener {

    private lateinit var titles: Array<String>
    private lateinit var drawerLayout: DrawerLayout
    private lateinit var drawerList: ListView

    val firebaseViewModel: FirebaseViewModel by viewModels()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_profile)

        titles = resources.getStringArray(R.array.titles)
        drawerLayout = findViewById(R.id.profile_drawer_layout)
        drawerList = findViewById(R.id.profile_drawer)
        drawerList.adapter = ArrayAdapter(
            this ,
            android.R.layout.simple_list_item_1 ,
            titles
        )
        drawerList.onItemClickListener = this
        val ft = supportFragmentManager.beginTransaction()
        ft.replace(R.id.profile_fragment_container , ProfileFragment())
        ft.setTransition(FragmentTransaction.TRANSIT_FRAGMENT_FADE)
        ft.commit()
    }

    override fun onStart() {
        super.onStart()
        if (!firebaseViewModel.isAuthenticated()) {
            val intent = Intent(application , StartActivity::class.java)
            startActivity(intent)
        }
    }

    override fun onClick(view: View) {
        when (view.id) {
            R.id.profile_settings -> {
                supportFragmentManager.beginTransaction()
                    .replace(R.id.profile_fragment_container , EditProfileFragment())
                    .setTransition(FragmentTransaction.TRANSIT_FRAGMENT_FADE)
                    .addToBackStack(null)
                    .commit()
            }
            R.id.profile_menu_btn -> drawerLayout.openDrawer(GravityCompat.START)
            else -> {
            }
        }
    }

    override fun onBackPressed() {
        if (drawerLayout.isDrawerOpen(drawerList)) {
            drawerLayout.closeDrawer(drawerList)
        } else {
            super.onBackPressed()
        }
    }

    override fun onItemClick(
        adapterView: AdapterView<*>? ,
        view: View ,
        i: Int ,
        l: Long
    ) {
        val intent = when (titles[i]) {
            "Карта" -> Intent(this , MapsActivity::class.java)
            "Рядом" -> Intent(this , NearbyActivity::class.java)
            "Мои мероприятия" -> Intent(this , MyPartiesActivity::class.java)
            else -> return
        }
        startActivity(intent)
        drawerLayout.closeDrawer(drawerList)
    }
}